src/terms/hubbard.jl

```diff
@@ -107,51 +107,68 @@ Hubbard energy, following the Dudarev et al. (1998) rotationally invariant forma
 ```
 """
 struct Hubbard{T}
-    manifold::OrbitalManifold
-    U::T
-    function Hubbard(manifold::OrbitalManifold, U)
-        U = austrip(U)
-        new{typeof(U)}(manifold, U)
+    manifolds::Vector{OrbitalManifold}
+    U::Vector{T}
+    function Hubbard(manifolds::Vector{OrbitalManifold}, U)
+        U = austrip.(U)
+        new{typeof(U[1])}(manifolds, U)
+    end
+    function Hubbard(params::Dict)
+        Hubbard(collect(keys(params)), collect(values(params)))
+    end
+    function Hubbard(params::Pair)
+        Hubbard([params[1]], [params[2]])
     end
 end
 function (hubbard::Hubbard)(basis::AbstractBasis)
-    manifold = resolve_hubbard_manifold(hubbard.manifold, basis.model)
-
+    manifolds = [resolve_hubbard_manifold(manifold, basis.model) for manifold in hubbard.manifolds]
     projs, labels = atomic_orbital_projectors(basis)
-    labels, projectors_matrix = extract_manifold(manifold, projs, labels)
-    TermHubbard(manifold, hubbard.U, projectors_matrix, labels)
+    manifold_labels, projs_matrices = [], []
+    for manifold in manifolds
+        push!(manifold_labels, extract_manifold(manifold, projs, labels).manifold_labels)
+        push!(projs_matrices, extract_manifold(manifold, projs, labels).manifold_projectors)
+    end
+    TermHubbard(manifolds, hubbard.U, projs_matrices, manifold_labels)
```

Comment:
Don't call `extract_manifold` twice. It might be simpler to use a list of pairs instead of two lists for Hubbard:
- Careful with `[]` because it creates a `Vector{Any}`, which will lead to dynamic dispatch and thus bad performance.
- Might make it easier to `map` over things.